framework/src/test/java/io/github/kgress/scaffold/webelement/BaseWebElementTests.java

```diff
@@ -395,4 +402,109 @@ public void testExceptionHandledHasClass(){
         assertFalse(elementByCssSelector.hasClass("class"));
     }
+
+  // these tests use both TestBaseWebElement & MockBaseWebElement
+
+  private List<MockBaseWebElement> setupForFindElementsTests(By parentBySelector, By rootChildBySelector) {
+    final WebElement reservationCard_A = mock(WebElement.class);
+    // this mock webelement instance is necessary to distinguish from reservationCard, since the
+    // findElements method does an Object.equals to see if the instance is in the filtered list
+    // of requested items.
+    final WebElement rogueDiv = mock(WebElement.class);
+    final WebElement webElement = mock(WebElement.class);
+    when(reservationCard_A.getTagName()).thenReturn("div");
+    when(reservationCard_A.findElement(any(By.ByXPath.class))).thenReturn(webElement);
+    // order here is important, to validate that rogueDiv causes the index of the desired elements
+    // to be nonsequential
+    when(webElement.findElements(any(By.ByTagName.class))).thenReturn(
+        List.of(reservationCard_A, rogueDiv, reservationCard_A, reservationCard_A)
+    );
+    when(webElement.findElements(rootChildBySelector)).thenReturn(
+        List.of(reservationCard_A, reservationCard_A, reservationCard_A)
+    );
+    TestBaseWebElement parentElement = new TestBaseWebElement(parentBySelector);
+    when(parentElement.getRawWebElement()).thenReturn(webElement);
+    assertThat(parentElement).isNotNull();
+    // now assert Order of child elements is what we expect...
+    List<MockBaseWebElement> reservationCards =
+        parentElement.findElements(MockBaseWebElement.class, rootChildBySelector);
+    assertThat(reservationCards).isNotNull();
+    assertThat(reservationCards.size()).isEqualTo(3);
+    return reservationCards;
+  }
+
+  @Test
+  public void testBaseWebElement_findElements_by_xPath() {
+    final By parentBySelector = By.xpath("//div[@class='reservation-container']");
+    final By rootChildBySelector = By.xpath("//div[@class='reservation-card-container']");
+    final var reservationCards = setupForFindElementsTests(parentBySelector, rootChildBySelector);
+    IntStream.range(0, reservationCards.size()).forEach((i) -> {
+      switch (i) {
+        case 0:
+          final String aByString = AutomationUtils.getUnderlyingLocatorByString(
+              reservationCards.get(i).getBy());
+          assertThat(aByString).isEqualTo("//div[@class='reservation-card-container'][1]");
+          break;
+        case 1:
+          final String bByString = AutomationUtils.getUnderlyingLocatorByString(
+              reservationCards.get(i).getBy());
+          assertThat(bByString).isEqualTo("//div[@class='reservation-card-container'][3]");
+          break;
+        case 2:
+          final String cByString = AutomationUtils.getUnderlyingLocatorByString(
+              reservationCards.get(i).getBy());
+          assertThat(cByString).isEqualTo("//div[@class='reservation-card-container'][4]");
+          break;
```

Comment:
I was double checking this, xpath does actually respect what you're filtering on. So this test should be testing that the indices returned are 1,2,3 and not 1,3,4 like it is in nth-child.

See an example here, using `xpath[1]` and `xpath[2]` return the correct results (since it evaluates the xpath to get the list, then does indexing, unlike nth-child which only indexes).

Indices 1/2 are correct:
<img width="554" alt="xpath-1" src="https://user-images.githubusercontent.com/8595173/160182326-68830e66-57d8-4de1-b5b7-9899a862f845.png">

<img width="557" alt="xpath-2" src="https://user-images.githubusercontent.com/8595173/160182376-642b66b9-df9c-4cc3-a837-289f3cbb3be2.png">

Index 3 in this webpage (like the xpath code would return would do) actually indexes an element that's not there.
<img width="558" alt="xpath-3" src="https://user-images.githubusercontent.com/8595173/160182420-876d3f13-dcf0-49b0-ae73-b42a926484fc.png">

Reply:
doh! fixing

Reply:
Looks good thanks!